include/pinocchio/multibody/model-graph.hpp

```diff
@@ -0,0 +1,370 @@
+//
+// Copyright (c) 2024 INRIA
+//
+
+#ifndef __pinocchio_multibody_model_graph_hpp__
+#define __pinocchio_multibody_model_graph_hpp__
+
+#include "pinocchio/multibody/fwd.hpp"
+#include "pinocchio/multibody/joint/joint-generic.hpp"
+#include "pinocchio/multibody/model.hpp"
+#include "pinocchio/spatial/inertia.hpp"
+#include "pinocchio/spatial/se3.hpp"
+
+#include <Eigen/Core>
+
+#include <boost/graph/adjacency_list.hpp>
+#include <boost/graph/depth_first_search.hpp>
+#include <boost/graph/visitors.hpp>
+
+#include <vector>
+#include <string>
+#include <iostream>
+#include <unordered_map>
+
+namespace pinocchio
+{
+  namespace internal
+  {
+    template<typename Derived>
+    struct JointBaseGraph
+    {
+      std::string name;
+
+      JointBaseGraph(const std::string & n)
+      : name(n) {};
+    };
+
+    struct JointFixedGraph : public JointBaseGraph<JointFixedGraph>
+    {
+      JointFixedGraph()
+      : JointBaseGraph("") {};
+      JointFixedGraph(const std::string & n)
+      : JointBaseGraph(n) {};
+    };
+
+    struct JointRevoluteGraph : public JointBaseGraph<JointRevoluteGraph>
+    {
+      // rotation axis
+      Eigen::Vector3d axis;
+      // axis to flip around if needed
+      Eigen::Vector3d flip_axis;
+      // if axis is unitary (X, Y, Z) and negative, then flipped is true, to remember to flip
+      // position when adding joint to model
+      bool flipped = false;
+      JointRevoluteGraph(const std::string & n, const Eigen::Vector3d & ax, const bool flip = false)
+      : JointBaseGraph(n)
+      , axis(ax)
+      , flipped(flip)
+      {
+        if (axis.isApprox(-Eigen::Vector3d::UnitX()))
+        {
+          flipped = true;
+          axis = Eigen::Vector3d::UnitX();
+          flip_axis = Eigen::Vector3d::UnitY();
+        }
+        else if (axis.isApprox(-Eigen::Vector3d::UnitY()))
+        {
+          flipped = true;
+          axis = Eigen::Vector3d::UnitY();
+          flip_axis = Eigen::Vector3d::UnitX();
+        }
+        else if (axis.isApprox(-Eigen::Vector3d::UnitZ()))
+        {
+          flipped = true;
+          axis = Eigen::Vector3d::UnitZ();
+          flip_axis = Eigen::Vector3d::UnitY();
+        }
+      };
```

Comment:
```suggestion
      }
```
You don't need to put ; after a function/method definition. You can remove it in all the file.